gson/src/test/java/com/google/gson/stream/JsonReaderTest.java

```diff
@@ -173,7 +173,22 @@ public void testHelloWorld() throws IOException {
     reader.endObject();
     assertEquals(JsonToken.END_DOCUMENT, reader.peek());
   }
-
+  
+  public void testInvalidJsonInput() throws IOException {
+	  String json = "{\n" +
+		        "   \"h\\ello\": true,\n" +
+		        "   \"foo\": [\"world\"]\n" +
+		        "}";
+
+	  try {
+		  JsonReader reader = new JsonReader(reader(json));
+		  reader.beginObject();
+		  assertEquals("hello", reader.nextName());
```

Comment:
the assertEquals() isn’t necessary, just calling nextName() should trigger it to throw